src/self_driving/planning/pilot_util.cpp

```diff
@@ -419,4 +425,233 @@ void PilotUtil::GroupFeaturesByOU(
   }
 }
 
+void PilotUtil::ComputeTableSizeRatios(const WorkloadForecast *forecast,
+                                       common::ManagedPointer<task::TaskManager> task_manager,
+                                       common::ManagedPointer<transaction::TransactionManager> txn_manager,
+                                       common::ManagedPointer<catalog::Catalog> catalog,
+                                       pilot::MemoryInfo *memory_info) {
+  NOISEPAGE_ASSERT(task_manager, "ComputeTableSizeRatios() requires task manager");
+
+  // Maps from table oid to the number of rows (acquired from pg_statistic)
+  std::unordered_map<catalog::table_oid_t, uint64_t> table_sizes;
+
+  auto to_row_fn = [&table_sizes, memory_info](const std::vector<execution::sql::Val *> &values) {
+    auto table_oid = static_cast<execution::sql::Integer *>(values[0])->val_;
+    auto num_rows = static_cast<execution::sql::Integer *>(values[1])->val_;
+
+    table_sizes[catalog::table_oid_t(table_oid)] = num_rows;
+    // Currently only track tables with more than 0 rows in the stats
+    memory_info->table_oids_.emplace_back(table_oid);
+  };
+
+  // Get <table_id, num_rows> pairs with num_rows > 0
+  auto query = fmt::format(
+      "select starelid, max(stanumrows) as num_rows from pg_statistic group by starelid "
+      "having max(stanumrows) > 0;");
+
+  common::Future<task::DummyResult> sync;
+  task_manager->AddTask(std::make_unique<task::TaskDML>(catalog::INVALID_DATABASE_OID, query,
```

Comment:
I think this is a bad query as `pg_statistic` lives inside of each `DatabaseCatalog` instance (or should based off the `pg_*` prefix).  If this does succeed, I'm curious as to which database/table is queried in the presence of multiple databases (and not just the default one created on first launch).

Reply:
I figured out why this can work. Essentially this query [goes to the `DEFAULT_DATABASE`](https://github.com/cmu-db/noisepage/blob/master/src/util/query_exec_util.cpp#L59-L67), and right now we're inserting everything into the `DEFAULT_DATABASE` when we're executing TPC-C under OLTPBench. Nevertheless, it just works coincidently and I'll fix it to be the correct `db_oid`.